airbyte-cdk/python/airbyte_cdk/sources/declarative/schema/json_file_schema_loader.py

```diff
@@ -15,11 +15,16 @@
 
 
 def _default_file_path() -> str:
-    # schema files are always in "source_<connector_name>/schemas/<stream_name>.json
-    # the connector's module name can be inferred by looking at the modules loaded and look for the one starting with source_
+    # Schema files are always in "source_<connector_name>/schemas/<stream_name>.json
+    # The connector's module name can be inferred by looking at the modules loaded and look for the one starting with source_
+    # One exception to this rule are reads invoked from the connector builder server which is prefixed by connector_builder
     source_modules = [
-        k for k, v in sys.modules.items() if "source_" in k  # example: ['source_exchange_rates', 'source_exchange_rates.source']
+        k
+        for k, v in sys.modules.items()
+        if "source_" in k or "connector_builder" in k  # example: ['source_exchange_rates', 'source_exchange_rates.source']
```

Comment:
this feels very smelly -- we don't expect the connector builder to have any schemas specific to a source, right? what is the right solution here?

Reply:
alternatively in the DefaultSchemaLoader we could just pass in a dummy filepath like `"/"` to skip assigning the default file path which was originally triggering the error. Avoids the issue you mentioned above. Or we decouple the default schema loader w/o reusing any part of the json schema loader

Reply:
Thinking about this more, I'm actually not sure what the right solution is. The root of the problem is that the DefaultSchemaLoader is currently trying to support two types of default behavior:

Scenario 1:
A connector developer for ease of use shouldn't need to define a JsonFileSchemaLoeader. We infer a default loader component that performs schema lookup based on the default file path for low-code sources.

Scenario 2:
When processing requests to read records from a manifest, the connector builder server wants to skip any attempt to retrieve a schema and use the empty schema (`{}`) as the default.

The problem is regardless of who is invoking/creating the loader, we need some kind of placeholder URL. I think a fair balance here is actually to build the default as we have it already if we find a source_module, but otherwise just attempt a search using the following template: `f"./{{{{options['name']}}}}.json"`. That allows us to skip hardcoding a module check for `connector_builder` which is the crux of the concerning code. We'll still attempt and probably always find nothing and that will defer to `{}` which is what we want for the connector builder.